src/olympia/reviewers/views.py

```diff
@@ -1426,8 +1426,10 @@ def get_version_object(self):
         if not hasattr(self, 'version_object'):
             self.version_object = get_object_or_404(
                 # The serializer will not need any of the stuff the
-                # transformers give us for the version.
-                self.get_addon_object().versions.all().no_transforms(),
+                # transformers give us for the version. We do need to fetch
+                # using an unfiltered manager to see deleted versions, though.
+                self.get_addon_object().versions(
+                    manager='unfiltered_for_relations').all().no_transforms(),
```

Comment:
can you point to the permission check/gating that limits this to just users with `Addons:ViewDeleted`?

Reply:
See `_verify_object_permissions()` a few lines after.